hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -131,7 +131,8 @@ public MapContainer(final String name, final Config config, final MapServiceCont
         this.queryEntryFactory = new QueryEntryFactory(mapConfig.getCacheDeserializedValues());
         this.objectNamespace = MapService.getObjectNamespace(name);
         initWanReplication(nodeEngine);
-        this.extractors = new Extractors(mapConfig.getMapAttributeConfigs(), config.getClassLoader());
+        ClassLoader classloader = mapServiceContext.getNodeEngine().getConfigClassLoader();
```

Comment:
this is the important change: use the classloader from node engine and not the one from the config.
a classloader in the config is from user configuration and obviously it does not take user code deployment into account.